clients/fides-js/src/lib/consent-utils.ts

```diff
@@ -0,0 +1,56 @@
+import { ConsentBannerOptions } from "./consent-types";
+
+/**
+ * Configuration options used for the consent banner. The default values (below)
+ * will be mutated by the banner() function to override with any user-provided
+ * options at runtime.
+ *
+ * This is effectively a global variable, but we provide getter/setter functions
+ * to make it seem safer and only export the getBannerOptions() one outside this
+ * module.
```

Comment:
The naming of this `globalBannerOptions` var is misleading because some of these options are more generic- e.g. geolocation, privacy center URL, and are needed at a parent level. Also we will still need a way to inject banner-specific options at runtime, so we can't delete it yet from being a global var. Unless there's a better pattern you see to handle this?

Reply:
> The naming of this globalBannerOptions var is misleading because some of these options are more generic- e.g. geolocation, privacy center URL, and are needed at a parent level. 

Yeah, that makes sense - if it's not banner-specific, let's just call them... `options`? We could design the options object to be something like:
```
options = {
  overlay: {
    ...
  },
  fidesApiUrl: "...",
  isGeolocationEnabled: true,
  ...
}
```

Having a shared `options` object is definitely reasonable for the overall `Fides.js` package, and instead of being in some clever state in this file it could just be a field on the `window.Fides` global

Reply:
See this comment too: https://github.com/ethyca/fides/pull/3191#discussion_r1187857126